packages/expo-updates/ios/EXUpdates/Update/RawManifest/EXUpdatesBaseRawManifest.m

```diff
@@ -0,0 +1,151 @@
+    return silentLaunch && [@(YES) isEqual:silentLaunch];
+  }
+  return false;
+}
+
+- (BOOL)isUsingDeveloperTool {
+  NSDictionary *manifestDeveloperConfig = self.rawManifestJSON[@"developer"];
+  BOOL isDeployedFromTool = (manifestDeveloperConfig && manifestDeveloperConfig[@"tool"] != nil);
+  return (isDeployedFromTool);
+}
+
+- (nullable NSString *)userInterfaceStyle {
+  if (self.iosConfig && self.iosConfig[@"userInterfaceStyle"]) {
+    return self.iosConfig[@"userInterfaceStyle"];
+  }
+  return self.rawManifestJSON[@"userInterfaceStyle"];
+}
+
+- (nullable NSString *)androidOrRootBackroundColor {
+  if (self.iosConfig && self.iosConfig[@"backgroundColor"]) {
+    return self.iosConfig[@"backgroundColor"];
+  }
+  return self.rawManifestJSON[@"backgroundColor"];
+}
+
+- (nullable NSString *)iosSplashBackgroundColor {
+  return [[self class] getStringFromManifest:self.rawManifestJSON
+                                       paths:@[
+                                         @[@"ios", @"splash", @"backgroundColor"],
+                                         @[@"splash", @"backgroundColor"],
+                                       ]];
+}
+
+- (nullable NSString *)iosSplashImageUrl {
+  return [[self class] getStringFromManifest:self.rawManifestJSON
+                                       paths:@[
+                                         [UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad
+                                         ? @[@"ios", @"splash", @"tabletImageUrl"]
+                                         : @[],
+                                         @[@"ios", @"splash", @"imageUrl"],
+                                         @[@"splash", @"imageUrl"],
+                                       ]];
+}
+
+- (nullable NSString *)iosSplashImageResizeMode {
+  return [[self class] getStringFromManifest:self.rawManifestJSON
+                                       paths:@[
+                                         @[@"ios", @"splash", @"resizeMode"],
+                                         @[@"splash", @"resizeMode"],
+                                       ]];
+}
+
++ (NSString * _Nullable)getStringFromManifest:(NSDictionary *)manifest
+                                        paths:(NSArray<NSArray<const NSString *> *> *)paths
+{
+  for (NSArray<const NSString *> *path in paths) {
+    NSString *result = [[self class] getStringFromManifest:manifest path:path];
+    if (result) {
+      return result;
+    }
+  }
+  return nil;
+}
+
++ (NSString * _Nullable)getStringFromManifest:(NSDictionary *)manifest
+                                         path:(NSArray<const NSString *> *)path
+{
+  NSDictionary *json = manifest;
+  for (int i = 0; i < path.count; i++) {
+    BOOL isLastKey = i == path.count - 1;
+    const NSString *key = path[i];
+    id value = json[key];
+    if (isLastKey && [value isKindOfClass:[NSString class]]) {
+      return value;
+    }
+    json = value;
```

Comment:
should we bail out if `value` is not an NSDictionary?

Reply:
This was moved from `EXManagedAppSplashScreenConfigurationBuilder` verbatim so I'm hesitant to change it.